doc/documents/Examples_Tutorials/Examples_Tutorials.rst

```diff
@@ -0,0 +1,623 @@
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+After we successfully pass basic Caffe CIFAR 10 tutorial with minor changes, we obtain the following files for deployment:
+
+ - cifar10_small.prototxt – textual description of the graph which might be visualized as in figure below.
+
+ - cifar10_small.caffemodel.h5 – data of trained graph. Not necessary h5 format.
+
+ - mean file (Optional). Not used in this example. The only mean value for all pixels in all channels was used: 128
+
+.. image::  ../images/2_CIFAR10_graph.png
+   :align: center
+   :alt: CIFAR-10 Graph Visualization
+
+.. 
+
+To get access to model data, install following modules in the environment:
+
+.. code:: python
+
+   import caffe
+   import lmdb
+   import numpy as np
+..
+
+To transform Caffe representation of data into numpy arrays, load and initialize model by constructing Caffe Net object which takes prototext and caffemodel files as parameters.
+
+.. code:: python
+
+   classifier = caffe.Net("cifar10_small.prototxt", "cifar10_small.caffemodel.h5", caffe.TEST)
+
+   # Get key to the first output of net (here it’s a prob data)
+   out_key = classifier.outputs[0] 
+..
+
+To deploy the model correctly, analyze the two subsets of data which might be transformed to numpy arrays by native interface:
+
+1) Model parameters. 
+
+.. code:: python
+
+   for layer_name, data_obj in classifier.params.iteritems():
+      weights_np_arr = data_obj[0].data # conv weights is blob #0 
+      bias_np_arr = data_obj[1].data   # conv bias is blob #1
+      # further operations with data
+..
+
+2) Intermediate results
+
+.. code:: python
+
+   for blob_name, data_obj in classifier.blobs.iteritems():
+   data_np_arr = data_obj.data
+   # further operations with data
+..
+
+In Caffe, these objects are referred as blobs. While model parameters in this case are fixed in the inference time (after model had been trained), intermediate results vary from one model input to another. Therefore, intermediate blob content differs depending on the input you feed into net. 
+
+To update classifier object state, use the following command:
+
+.. code:: python
+
+    pred = classifier.forward_all(data=test_data)[out_key]
+..
+
+Here:
+
+ - `test_data` is a numpy array with input vector (CIFAR-10 dataset entity),
+ 
+ - `out_key` is our “key” to the network output we had defined early
+ 
+ - `pred` is the output.
+
+Using defined pieces of Python code, you can extract all the required data from the model and adapt it to an embedded MLI based application.  
+
+Collect Data Range Statistic for Each Layer
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Quantization process is not only meant to convert weights data to fixed point representation, but also meant to define ranges of all the intermediate data for each layer. For this purpose, run the model on some representative data subset and gather statistics for all intermediate results. It is better to use all training subsets, or even all the dataset. 
```

Comment:
"It is better to use all training subsets, or even all the dataset" -> "It is better to use all training subset."